src/server/game/Spells/Spell.cpp

```diff
@@ -3725,14 +3756,53 @@ void Spell::update(uint32 difftime)
             }
 
             if (m_timer == 0 && !m_spellInfo->IsNextMeleeSwingSpell() && !IsAutoRepeat())
+            {
                 // don't CheckCast for instant spells - done in spell::prepare, skip duplicate checks, needed for range checks for example
                 cast(!m_casttime);
+
+                // If the spell is an Instant Spell clear state casting.
+                if (!m_spellInfo->IsChanneled())
+                {
+                    Unit* unitCaster = m_caster->ToUnit();
+                    if (!unitCaster)
+                        return;
+
+                    unitCaster->ClearUnitState(UNIT_STATE_CASTING);
```

Comment:
Are you sure `UNIT_STATE_CASTING` is set with instant spells?
IIRC `UNIT_STATE_CASTING` is only sit on the Unit method `SetCurrentSpellOrWhateverTheNameIs`

Reply:
I checked it looks like generic instant spells with cast time get the state. So I think this can be removed as was a remnant from the previous patch.